src/core/errorHandler.js

```diff
@@ -140,6 +167,29 @@ export function withErrorHandler({ name, id }) {
 }
 
 /*
+ * This is a page level error decorator. It works like the `withErrorHandler()`
+ * decorator, but aims at synchronizing both the server and client sides and
+ * should be used for page level components.
+ *
+ * Pass the optional `extractId` function to create "per unique page" level
+ * error handlers. This function takes the component's props and must returns a
+ * unique value based on these props (e.g., based on the `slug`, `uniqueId`,
+ * etc.).
+ *
+ * When this optional function is not supplied, we "fix" the error handler `id`
+ * (e.g., `SomeComponentPage`) otherwise we pass the `name` suffixed by `Page`,
+ * and the HOC will create the final error handler ID thanks to `extractId`.
+ */
+export const withPageErrorHandler = ({ name, extractId = null }) => {
```

Comment:
Since we are introducing this as a new HOC, I suggest making `extractId` a mandatory parameter. If it's falsy, throw an error. If the component can't do `extractId: ...` then it should just use the old `withErrorHandler` HOC.